source/utils/adios_remote_read/adios_remote_read.cpp

```diff
@@ -0,0 +1,721 @@
+                        startZ = globalSizeZ / 2;
+
+                        if (ratio == 1.0)
+                        {
+                            startX = 0;
+                            countX = globalSizeX;
+                        }
+                        else
+                        {
+                            startX = (1 - ratio) / 2.0 * globalSizeX;
+                            countX = globalSizeX * ratio;
+                        }
+
+                        break;
+                    default:
+                        break;
+                    }
+                    var.SetSelection(adios2::Box<adios2::Dims>({startX, startY, startZ},
+                                                               {countX, countY, countZ}));
+
+                    size_t elementsSize = var.SelectionSize();
+                    std::vector<double> data2D(elementsSize);
+                    reader.Get<double>(var, data2D.data(), adios2::Mode::Sync);
+                }
+            }
+
+            reader.EndStep();
+        }
+        reader.Close();
+        end_time = std::chrono::system_clock::now();
+        std::chrono::duration<double> elapsed_seconds = end_time - start_time;
+        out += "Duration: " + std::to_string(elapsed_seconds.count()) + "\n";
+        std::ofstream out_file;
+        out_file.open("output3DPlane-" + std::to_string(rank) + ".log");
+        out_file << out;
+        out_file.close();
+    }
+    catch (std::invalid_argument &e)
+    {
+        if (rank == 0)
+        {
+            std::cout << "Invalid argument exception, STOPPING PROGRAM\n";
+            std::cout << e.what() << "\n";
+        }
+    }
+    catch (std::ios_base::failure &e)
+    {
+        if (rank == 0)
+        {
+            std::cout << "System exception, STOPPING PROGRAM\n";
+            std::cout << e.what() << "\n";
+        }
+    }
+    catch (std::exception &e)
+    {
+        if (rank == 0)
+        {
+            std::cout << "Exception, STOPPING PROGRAM\n";
+            std::cout << e.what() << "\n";
+        }
+    }
+}
+
+int main(int argc, char *argv[])
+{
+
+    int rank = 0;
+    int nproc;
+#if ADIOS2_USE_MPI
+    MPI_Init(&argc, &argv);
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &nproc);
+#endif
+    int nsteps = 1;
+
+#if ADIOS2_USE_MPI
+    adios2::ADIOS adios(MPI_COMM_WORLD);
+#else
+    adios2::ADIOS adios;
+#endif
+    // options
+    std::string filename;
+    std::string engine = "BP5";
+    std::string transport = "filesystem";
+    int mode = -1;
+    double ratio = 1.0;
+    int output_line_length = 10;
+
+    // should be adjusted for getopt
+    auto start = std::vector<size_t>(3);
+    auto count = std::vector<size_t>(3);
+    std::vector<std::string> variables;
+
+    option longopts[] = {{"help", no_argument, NULL, 'h'},
+                         {"case", required_argument, NULL, 'c'},
+                         {"filename", required_argument, NULL, 'f'},
```

Comment:
Do you create the data as part of the test or where do you expect to get the input file?

Reply:
It can be actually used for reading data generated by adios_iotest program or any other data file, say generated by the  S3D program.

Reply:
Ok, at some point we should add documentation on how to use this and examples of how you can generate the data for readthedocs